src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,848 @@
+                else :
+                    cephfs.unlink(to_bytes(file_path))
+        except :
+            traceback.print_exc(file = sys.stdout)
+   
+    def complete_mv(self, text, line, begidx, endidx):
+        """
+         auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    mv_parser = argparse.ArgumentParser(description = 'Move File.')
+    mv_parser.add_argument('src_path', type = str, help = 'Source File Path.')
+    mv_parser.add_argument('dest_path', type = str, help = 'Destination File Path.')
+
+    @with_argparser(mv_parser)
+    def do_mv(self, args):
+        """
+        Rename a file or Move a file from source path to the destination
+            Usage : mv <src_path> <dest_path>
+        """
+        try:           
+            cephfs.rename(to_bytes(args.src_path), to_bytes(args.dest_path))
+        except :
+            traceback.print_exc(file = sys.stdout)
+   
+    def complete_cd(self, text, line, begidx, endidx):
+        """ 
+        auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    cd_parser = argparse.ArgumentParser(description = 'Create Directory.')
+    cd_parser.add_argument('dir_name', type = str, help = 'Name of the directory.', nargs = '*', default = [''])
+
+    @with_argparser(cd_parser)
+    def do_cd(self, args):
+        """ 
+        Open a specific directory.
+            Usage : cd <dir_name>
+        """
+        try:            
+            if args.dir_name[0] == '' :
+                cephfs.chdir(b'/')
+            if args.dir_name[0] == '..':
+                dir_name = cephfs.getcwd().decode('utf-8').rsplit('/', 1)[0]
+                if dir_name != '':
+                    cephfs.chdir(to_bytes(dir_name))
+                else :
+                    cephfs.chdir(b'/')
+            else :
+                cephfs.chdir(to_bytes(args.dir_name[0]))
+            self.working_dir = cephfs.getcwd().decode('utf-8')
+            self.set_prompt()
+        except :
+            traceback.print_exc(file = sys.stdout)
+   
+    def do_cwd(self, arglist):
+        """
+        Get current working directory.
+           Usage : cwd
+        """
+        print(cephfs.getcwd())
+
+    def complete_chmod(self, text, line, begidx, endidx):
+        """
+        auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    chmod_parser = argparse.ArgumentParser(description = 'Create Directory.')
+    chmod_parser.add_argument('file_name', type = str, help = 'Name of the file')
+    chmod_parser.add_argument('mode', type = int, help = 'Mode')
+
+    @with_argparser(chmod_parser)
+    def do_chmod(self, args):
+        """
+        Change permission of a file
+        """
+        try:           
+            cephfs.chmod(args.file_name, args.mode)
+        except :
+             traceback.print_exc(file = sys.stdout)
```

Comment:
this pattern is repeating itself in this script. could make it a decorator also.